### PR TITLE
feat: cohortdefiners return validatedframes

### PR DESCRIPTION
**Pull request by sarakolding**
Automatically created PR from invoke

### CODE REVIEW

**Comment by sarakolding** [comment]
@MartinBernstorff what I think is happening is that the PredictionTimeCollator (in seq models) is no longer runtime_checkable since it contains CohortDefiner, and CohortDefiner now uses the ValidatedFrame which is not runtime_checkable. But I'm pretty lost (and Lasse is too), so maybe we can talk about it tomorrow 🐒

**Comment by sarakolding** [comment]
(ps it is not done - I can't run the tests because of the error)

**Comment by MartinBernstorff** [comment]
Sure thing. I'm here today; I'll wait until we can talk 👍

**Comment by sarakolding** [comment]
thoughts: more cols/validators? e.g., uuid col with a unique rule?
also, maybe we should make a getattribute method or something, so the frame can be called from the instance, instead of the attribute?

**Review by MartinBernstorff — CHANGES_REQUESTED** [review]

File: psycop/common/test_cohort_definition.py [hunk 1]
```diff
@@ -36,4 +36,4 @@ def apply(self, df: pl.LazyFrame) -> pl.LazyFrame:
         entity_id_col_name="entity_id",
```
[the remark this answers is not in the document]
Comment (sarakolding) [reply psycop/common/test_cohort_definition.py:36]: works because it is only used for counting, and get_counts=False

**Review by MartinBernstorff — APPROVED** [review]

**Pull request merged**